Mathlib/Data/List/Infix.lean

```diff
@@ -469,7 +469,8 @@ theorem insert_nil (a : α) : insert a nil = [a] :=
   rfl
 #align list.insert_nil List.insert_nil
 
-theorem insert.def (a : α) (l : List α) : insert a l = if a ∈ l then l else a :: l :=
+@[deprecated]
+theorem insert.def (a : α) (l : List α) : insert a l = if l.elem a then l else a :: l :=
```

Comment:
What's this deprecated in favor of? I think it would be worth keeping the original lemma statement around, even if it's no longer `rfl` and renamed to `insert_eq_ite`

Reply:
Renamed to `insert_eq_ite`.